### DISCUSSION TITLE
kekulize mol question

### DISCUSSION THREAD

**Discussion by tsn2027** [rdkit/rdkit#7193] [Q&A] [answered]
Hello all,

I am trying to make ```C=C1[CH]C=CC=C1``` from ```c1ccccc1[CH2]```

```
radical_smi = "c1ccccc1[CH2]"
mol = Chem.MolFromSmiles(radical_smi)
# mol = Chem.AddHs(mol) # unsure whether adding H would help.
products = AllChem.ReactionFromSmarts("[Cv3:0][c:1][cv4:2]>>[C:0]=[c:1][cv3:2]").RunReactants((mol,))

for i in range(len(products)):
    i1 = products[i][0]
    Chem.SanitizeMol(i1)
    display(Draw.MolToImage(i_1,size=(300,100)))
```

However, I received this error:
```
Can't kekulize mol.  Unkekulized atoms: 2 3 4 5 6
---------------------------------------------------------------------------
KekulizeException                         Traceback (most recent call last)
KekulizeException: Can't kekulize mol.  Unkekulized atoms: 2 3 4 5 6
```

If I don't kekulize mol, I got a weird ring connection, and a product does not look right either.

Does anyone know how to fix this problem or have an alternative approach? Thanks!

**Comment by dehaenw** [comment, marked as answer]
In my experience it is quite hard to work with radicals using the runreactants machinery.
Here is one approach that gives the intended output, the approach here is to use explicit Hs during the reaction, and then remove them afterwards. Another factor is that the entire benzene ring is factored in and explicitly dearomatized. See the below codeblock, which output the two expected structures
```python
radical_smi = "c1cccc(Cl)c1[CH2]"
mol = Chem.MolFromSmiles(radical_smi)
mol = Chem.AddHs(mol) # unsure whether adding H would help.
products = AllChem.ReactionFromSmarts("[Cv3:0]-[c:1]1[c:2][c:3][c:4][c:5][c:6]1>>[C:0]=[C:1]1-[Cv3:2]=[C:3]-[C:4]=[C:5]-[C:6]1").RunReactants((mol,))

for i in range(len(products)):
    i1 = products[i][0]
    i1 = Chem.RemoveAllHs(i1)
    display(Draw.MolToImage(i1,size=(300,100)))
```
![image](https://github.com/rdkit/rdkit/assets/66372095/0c3c7127-74ee-4ff3-acbb-2e546f240fb3)
![image](https://github.com/rdkit/rdkit/assets/66372095/f7e8d931-51bc-49f9-950f-b8e7fa71ec9c)

**Reply by tsn2027** [reply]
Thank you! This is what I want to get!
